src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const separatorInner = separator + '';
+    let res = '';
+
+    for (const item of this) {
+      if (item !== null && item !== undefined) {
+        res += item;
+      }
+
+      res += separator;
+    }
+
+    if (separatorInner || separatorInner.length) {
```

Comment:
this condition is always `true`, since you have a default value for `separator `

Reply:
ok, in your solution we can't avoid if and slice. failed test with empty separator
also you can instead for of loop make with for loop and check current index. if it is last element - do not add a separator

Reply:
or another solution
remove if and return with slice
```
 return res
      .slice(0, -separatorInner.length || this.length);